src/Topics/TopicsPage.jsx

```diff
@@ -0,0 +1,34 @@
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+
+
+const TopicsPage = ({ articles }) => {
+  const [articleByTopic, setArticleByTopic] = useState([]);
+  const { topic_slug } = useParams();
+
+  useEffect(() => {
+    if (topic_slug) {
+      const filteredArticles = articles.filter(
+        (article) => article.topic === topic_slug
+      );
+      setArticleByTopic(filteredArticles);
```

Comment:
You're just filtering in the front end here which you shouldn't need to do. You can reuse your API call of `getArticles`, then pass in an optional parameter of `topic_slug` using axios params. This will mean the API function will fetch the articles by topic if one is defined, if not it would bring all of the articles.
It would also mean that you wouldn't need this TopicsPage and could render the articles in ArticleList, with the back end deciding whether or not it brings all the articles or just the ones based on a topic. 👍